tests/lean/withSetOptionIn.lean

```diff
@@ -0,0 +1,32 @@
+import Lean
+
+/-!
+# `withSetOptionIn`
+
+This test checks that `withSetOptionIn` recurses into the command syntax (`stx[2]`) in
+`set_option ... in <cmd>`.
+
+Prior to #3806, `withSetOptionIn` erroneously recursed into the syntax `in` (`stx[1]`).
```

Comment:
Since the PR description already mentions this, maybe this line can be removed?

Reply:
I'd like the info to be immediately available to anyone reading the test without having to go to github! :)